clippy_lints/src/methods/utils.rs

```diff
@@ -142,8 +143,12 @@ impl<'cx, 'tcx> Visitor<'tcx> for CloneOrCopyVisitor<'cx, 'tcx> {
         if self.is_binding(expr) {
             if let Some(parent) = get_parent_expr(self.cx, expr) {
                 match parent.kind {
-                    ExprKind::AddrOf(BorrowKind::Ref, Mutability::Not, _) => {
-                        self.addr_of_exprs.push(parent);
+                    ExprKind::AddrOf(BorrowKind::Ref, Mutability::Not, referent) => {
+                        if !parent.span.from_expansion()
+                            && let Ok(snippet) = self.cx.sess().source_map().span_to_snippet(referent.span)
```

Comment:
I'm not sure if this is the best way to convert the span to source code. Feel free to suggest a more suitable way.

Reply:
That could work, but we can also take the span of the `&` with `parent.span.with_hi(referent.lo())` [like before](https://github.com/rust-lang/rust-clippy/blob/5aae5f6ae6b4ba0e6ddadea43e6a2bcadcaa2328/clippy_lints/src/methods/unnecessary_iter_cloned.rs#L129) and use the empty string for the suggestion (to remove it). That would probably make this a bit simpler